crates/api/src/local_user/login.rs

```diff
@@ -36,11 +31,13 @@ pub async fn login(
       .ok_or(LemmyErrorType::IncorrectLogin)?;
 
   // Verify the password
-  let valid: bool = verify(
-    &data.password,
-    &local_user_view.local_user.password_encrypted,
-  )
-  .unwrap_or(false);
+  let valid: bool = if let Some(password_encrypted) = &local_user_view.local_user.password_encrypted
+  {
+    verify(&data.password, password_encrypted).unwrap_or(false)
+  } else {
+    false
```

Comment:
This should use a separate error type called something like `IncorrectLoginMethod` or `NoPasswordSet`

Reply:
This would be better:

```rust
  let valid: bool = local_user_view
    .local_user
    .password_encrypted.clone()
    .and_then(|password_encrypted| {
      verify(&data.password, &password_encrypted).ok()
    })
    .unwrap_or(false);
```